src/charm.py

```diff
@@ -481,159 +45,60 @@ def __init__(self, *args) -> None:  # type: ignore[no-untyped-def]
-            all_relations = (None,)  # type: ignore[assignment]
-        multiple_rels = self._multiple_relations
-        return [
-            _ConfigOrRelation(self.model, self.config, relation, multiple_rels)
-            for relation in all_relations
-        ]
-
-    @property
-    def _multiple_relations(self) -> bool:
-        """Check if we're related to multiple applications.
-
-        Returns:
-            if we're related to multiple applications or not.
-        """
-        return len(self._deduped_relations()) > 1
-
-    @property
-    def _namespace(self) -> Any:
-        """Namespace for this ingress.
-
-        Returns:
-            The namespace for this Ingress.
-        """
-        # We're querying the first one here because this will always be the same
-        # for all instances. It would be very unusual for a relation to specify
-        # this (arguably we should remove this as a relation option), so if set
-        # via config it will be the same for all relations.
-        return self._all_config_or_relations[0]._namespace
+            The current effective relation object, None if there are no relation.
+        """
+        if self.model.get_relation("nginx-route") is not None:
+            relation = cast(Relation, self.model.get_relation("nginx-route"))
+            if not (relation.app is None or not relation.data[relation.app]):
+                return relation
+        elif self.model.get_relation("ingress") is not None:
+            relation = cast(Relation, self.model.get_relation("ingress"))
+            if not (relation.app is None or not relation.data[relation.app]):
+                return relation
+        return None
 
-    def _describe_ingresses_action(self, event: Any) -> None:
-        """Handle the 'describe-ingresses' action.
+    def _get_definition_from_relation(self, relation: Relation) -> IngressDefinition:
+        """Get the IngressDefinition from the given relation.
 
         Args:
-            event: Juju event that fires this handler.
+            relation: The source relation object.
+
+        Return:
+            The IngressDefinition corresponding to the provided relation.
         """
-        api = self._networking_v1_api()
-        ingresses = api.list_namespaced_ingress(namespace=self._namespace)
-        event.set_results({"ingresses": ingresses})
+        if relation.name == "nginx-route":
+            definition_essence = IngressDefinitionEssence(
+                model=self.model, config=self.config, relation=relation
+            )
+        else:
+            definition_essence = IngressDefinitionEssence(
+                model=self.model,
+                config=self.config,
+                relation=relation,
+                ingress_provider=self._ingress_provider,
+            )
+        ingress_definition = IngressDefinition.from_essence(definition_essence)
+        return ingress_definition
 
-    def k8s_auth(self) -> None:
+    def _k8s_auth(self) -> None:
```

Comment:
Let's move this to the constructor

Reply:
`kubernetes.config.load_incluster_config` has been moved to the charm constructor.